src/widgets/geo-search/__tests__/geo-search.test.tsx

```diff
@@ -0,0 +1,362 @@
+/**
+ * @jest-environment jsdom
+ */
+import { initialize } from '@googlemaps/jest-mocks';
+import { createSearchClient } from '../../../../test/mock/createSearchClient';
+import instantsearch from '../../../index.es';
+import { wait } from '../../../../test/utils/wait';
+import geoSearch from '../geo-search';
+import {
+  createMultiSearchResponse,
+  createSingleSearchResponse,
+} from '../../../../test/mock/createAPIResponse';
+import type { SearchResponse } from '../../../../src/types';
+
+beforeEach(() => {
+  document.body.innerHTML = '';
+  initialize();
+});
+
+describe('geoSearch', () => {
+  describe('templates', () => {
+    test('does not warn with default templates', async () => {
+      const container = document.createElement('div');
+      const searchClient = createMockedSearchClient();
+
+      const search = instantsearch({
+        indexName: 'indexName',
+        searchClient,
+      });
+
+      search.addWidgets([
+        geoSearch({
+          container,
+          // eslint-disable-next-line no-undef
+          googleReference: google,
+        }),
+      ]);
+
+      // @MAJOR Once Hogan.js and string-based templates are removed,
+      // `search.start()` can be moved to the test body and the following
+      // assertion can go away.
+      expect(async () => {
+        search.start();
+
+        await wait(0);
+      }).not.toWarnDev();
+
+      await wait(0);
+    });
+
+    type CustomHit = { name: string; description: string };
+
+    function createMockedSearchClient(
```

Comment:
this could be top-level

Reply:
I nested it in the `describe` block for all other widget tests, partly because this prevented having to make a more generic function right away.

Would it be worth moving them all as we write more tests in those files instead?

Reply:
Personally I prefer having all the setup functions top-level as it's impossible to accidentally read over tests that way. I'm ok with doing this later too though

Reply:
I'll do this in the final PR, once all stacked ones are merged.